### ISSUE TITLE
Troubles with VSCodeVim extension

### ISSUE THREAD

**Issue by MirKml** [CFiggers/vscode-janet-plus-plus#25]
Hello, I'm long time user of VSCodeVim ext - http://aka.ms/vscodevim, and I'm trying to use also your Janet++ ext, mostly because of I'm plan to use power of s-expr editing.

But it seems that Janet++ has some serious troubles with VSCodeVim - when I type comment it makes another EOL.
I can make a better error report, but initial question - is this combination supported - VSCodeVim, Janet++?
From Calva point of view, it is, but will some small issues https://calva.io/vim/. And Calva is base for your ext as I understand.

**Comment by CFiggers** [comment]
Hey there—nice to hear from someone who's using Janet++ and enjoying it. 

I would not say that the interaction between Janet++ and VSCodeVim is "supported" in the sense that anyone (aside from yourself, that I know of) is actively testing the two together in order to find issues ahead of time. But I don't mind trying to figure out bugs and making fixes to make the interaction between the two smoother.

If you want to go ahead and send in a more detailed bug report and repro steps, feel free. I am the only maintainer and I haven't focused on Janet++ in a little while, so no promises for timeframe. But when I get some time I'll have a look.

**Comment by MirKml** [comment]
hello @CFiggers, I have the detailed reproduction
tl;dr - no vim command - e.g. ":w" for saving - doesn't work in "comment area" - every line after `#` character

Detailed description:
Install VSCodeVim http://aka.ms/vscodevim extension and your extension installed, without any settings modification.

When I have some .janet file, I make some comment. e.g. `# test comment`, in vim insert mode, hit <kbd>ESC</kbd> back to normal mode, this is the result - correct
![image](https://github.com/user-attachments/assets/a6ad413a-b77d-4bf7-91d0-e63c6c9602b5)
then I type some vim command e.g. `:w` for saving, and hit <kbd>ENTER</kbd> for executing such vim command and
there is a problem
- desired vim command isn't executed at all - e.g. file isn't saved, string searched .....
- new EOL is inserted into the document instead

this is the result
![image](https://github.com/user-attachments/assets/19e207d4-3f14-4088-ba9a-ccebfdd85ced)

This is annoying, because in "comment area", no vim commands works as expected, even vim search with `/<pattern>`, inserted EOLs breaks the coding focus and needs to be removed.
I have to move into "code area", then vim commands works as expected.
Try to fix please, your ext is very useful.

**Comment by auxym** [comment]
The indentation also does not seem to play well with the Vim extension. For example if I have this (pipe showing my cursor, in insert mode):

```janet
(def input (readfile input-path)|)
```

And hit enter for a new line, I get this:

```janet
(def input (readfile input-path)
|  )
```

So the cursor is at the beginning of the line, and I have to manually move right by two spaces.
